File: 56.Merge Intervals/solution.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        if not intervals:
            return []
        intervals.sort()
        merger = []
        merger.append(intervals[0])

        for interval in intervals[1:]:
            if interval[0] <= merger[-1][1]:
                if interval[1] >= merger[-1][1]:
                    merger[-1][1] = interval[1]
            else:
                merger.append(interval)
        return merger
        

File: 56.Merge Intervals/solution.cpp
class Solution {
public:
    bool static cmp(vector<int>&a, vector<int>&b){
        return a[0]<b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end(),cmp);
        vector<vector<int>> res;
        res.push_back(intervals[0]);
        int n = intervals.size();
        for(int i = 1; i < n; i++) {
            if (intervals[i][0] <= res[res.size()-1][1]) {
                res[res.size()-1][1] = max(res[res.size()-1][1], intervals[i][1]);
            }else{
                res.push_back(intervals[i]);
            }
        }
        return res;
    }
};